src/group_theory/complement.lean

```diff
@@ -0,0 +1,296 @@
+/-
+Copyright (c) 2021 Thomas Browning. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Thomas Browning
+-/
+
+import group_theory.group_action
+import group_theory.order_of_element
+import group_theory.quotient_group
+
+/-!
+# Complements
+
+In this file we define the complement of a subgroup.
+
+## Main definitions
+
+- `is_complement S T` where `S` and `T` are subsets of `G`
+- `left_transversals T` where `T` is a subset of `G`
+- `right_transversals S` where `S` is a subset of `G`
+
+## Main results
+
+- `is_complement_of_coprime` : Subgroups of coprime order are complements
+- `schur_zassenhaus` : Schur-Zassenhaus for abelian normal subgroups
+-/
+
+open_locale big_operators
+
+namespace subgroup
+
+variables {G : Type*} [group G] (H K : subgroup G) (S T : set G)
+
+/-- `S` and `T` are complements if `(*) : S × T → G` is a bijection -/
+@[to_additive "`S` and `T` are complements if `(*) : S × T → G` is a bijection"]
+def is_complement : Prop := ∀ g : G, ∃! x : S × T, x.1.1 * x.2.1 = g
```

Comment:
Why did you choose to write this with `exists_unique` statement over `function.bijective`? From reading the rest of the file, it seems you need to do more translation between the `exists_unique` and `function.bijective`, than you get to apply `exists_unique` directly.

Reply:
I have changed this